amethyst_network/src/network_socket.rs

```diff
@@ -89,9 +89,12 @@ where
         let mut socket = UdpSocket::bind(addr, NetworkConfig::default())
             .map_err(|x| Error::new(ErrorKind::Other, x.to_string()))?;
 
-        socket
-            .set_nonblocking(true)
-            .expect("Unable to set `UdpSocket` to non-blocking mode");
+        socket.set_nonblocking(true).map_err(|_| {
+            Error::new(
+                ErrorKind::Other,
+                "Unable to set `UdpSocket` to non-blocking mode".to_owned(),
```

Comment:
`.to_owned()` is not necessary, all though it is worth noting that it still [does convert it into a String internally](https://doc.rust-lang.org/src/std/error.rs.html#271).

Reply:
Hmmm... I could have sworn it wasn't letting me pass in a string slice. Must have been from when I was trying something else. I'll remove it.

I don't think it matters so much about allocating the String, as it only happens on the non-happy path.